modin/pandas/test/utils.py

```diff
@@ -573,7 +584,93 @@ def assert_empty_frame_equal(df1, df2):
         assert False, f"Empty frames have different types: {type(df1)} != {type(df2)}"
 
 
-def df_equals(df1, df2):
+def assert_all_act_same(condition, *objs):
+    """
+    Assert that all of the objs give the same boolean result for the passed condition (either all True or all False).
+
+    Parameters
+    ----------
+    condition : callable(obj) -> bool
+        Condition to run on the passed objects.
+    *objs :
+        Objects to pass to the condition.
+
+    Returns
+    -------
+    bool
+        Result of the condition.
+    """
+    results = [condition(obj) for obj in objs]
+    if len(results) < 2:
+        return
+
+    assert all(results[0] == res for res in results[1:])
+    return results[0]
+
+
+def _maybe_cast_to_pandas_dtype(dtype):
+    """Cast passed `dtype` to according pandas dtype if needed for the sake of equality comparison."""
+    # If we're running in a cloud mode then all the numpy types are substituted with a proxy net-reference,
+    # Such dtypes won't pass equality check with pandas dtypes thus manually converting them to a pure numpy
+    if "netref" in str(type(dtype)):
+        return np.dtype(dtype.name)
+    return dtype
+
+
+def assert_dtypes_equal(df1, df2):
+    """
+    Assert that the two passed DataFrame/Series objects have equal dtypes.
+
+    The function doesn't require that the dtypes are identical, it has the following reliefs:
+        1. The dtypes are not required to be in the same order
+           (e.g. {"col1": int, "col2": float} == {"col2": float, "col1": int})
+        2. The dtypes are only required to be in the same class
+           (e.g. both numerical, both categorical, etc...)
```

Comment:
Should we remove this relaxation in the future and check more stringently?

Reply:
same question from my side